File: app/src/main/java/com/unistay/android/domain/model/model/ChatSession.kt
package com.unistay.android.domain.model.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "chat_sessions")
data class ChatSession(
    @PrimaryKey val sessionId: String,
    val studentId: String,
    val landlordId: String,
    val roomId: String,
    val roomTitle: String,
    val landlordName: String,
    val landlordPhone: String,
    val lastMessage: String,
    val studentUnreadCount: Int = 0,
    val landlordUnreadCount: Int = 0,
    val lastUpdated: Long = System.currentTimeMillis(),
    val studentAvatarUrl: String? = null,
    val landlordAvatarUrl: String? = null,
    val studentName: String? = null
)

File: UniStay.Api/Models/ChatSession.cs
using System.ComponentModel.DataAnnotations;

namespace UniStay.Api.Models
{
    public class ChatSession
    {
        [Key]
        public string SessionId { get; set; } = string.Empty;
        public string StudentId { get; set; } = string.Empty;
        public string LandlordId { get; set; } = string.Empty;
        public string RoomId { get; set; } = string.Empty;
        public string RoomTitle { get; set; } = string.Empty;
        public string LandlordName { get; set; } = string.Empty;
        public string LandlordPhone { get; set; } = string.Empty;
        public string LastMessage { get; set; } = string.Empty;
        public long LastUpdated { get; set; }
        public int StudentUnreadCount { get; set; } = 0;
        public int LandlordUnreadCount { get; set; } = 0;

        [System.ComponentModel.DataAnnotations.Schema.NotMapped]
        public string? StudentAvatarUrl { get; set; }
        [System.ComponentModel.DataAnnotations.Schema.NotMapped]
        public string? LandlordAvatarUrl { get; set; }
        [System.ComponentModel.DataAnnotations.Schema.NotMapped]
        public string? StudentName { get; set; }
    }
}
